src/object/srv_ec_aggregate.c

```diff
@@ -291,62 +289,27 @@ static void
 agg_clear_extents(struct ec_agg_entry *entry)
 {
 	struct ec_agg_extent	*extent, *ext_tmp;
-	uint64_t		 tail, ptail = 0U;
+	uint64_t		 tail;
 	bool			 carry_is_hole = false;
 
-	if (entry->ae_cur_stripe.as_ho_ext_cnt) {
-		d_list_for_each_entry_safe(extent, ext_tmp,
-					   &entry->ae_cur_stripe.as_hoextents,
-					   ae_link) {
-			d_list_del(&extent->ae_link);
-			D_FREE_PTR(extent);
-		}
-		entry->ae_cur_stripe.as_ho_ext_cnt = 0;
-	}
-
 	d_list_for_each_entry_safe(extent, ext_tmp,
 				   &entry->ae_cur_stripe.as_dextents,
 				   ae_link) {
-		unsigned long next_stripe_st =
-			(entry->ae_cur_stripe.as_stripenum + 1) *
-				 ec_age2ss(entry);
-
 		/* Check for carry-over extent. */
 		tail = agg_carry_over(entry, extent);
+		D_ASSERT(tail == 0);
```

Comment:
oh, the tail will be cut off, see agg_data_extent()

if (this_stripenum == agg_entry->ae_cur_stripe.as_stripenum &&
 	    this_end_stripenum > agg_entry->ae_cur_stripe.as_stripenum) {
 		new_recx.rx_nr = (this_stripenum + 1) * ec_age2ss(agg_entry) -
 				  entry->ie_recx.rx_idx;
 		rc = agg_extent_add(agg_entry, entry, &new_recx);
 		if (rc)
 			return rc;
 	}
So there should not be any cross-stripe extent at the tail.

Reply:
I know, but before that code you pasted, with such code ahead:
1967 >       if (agg_entry->ae_cur_stripe.as_extent_cnt == 0 ||                      
1968 >           (this_stripenum == agg_entry->ae_cur_stripe.as_stripenum &&         
1969 >           this_end_stripenum == agg_entry->ae_cur_stripe.as_stripenum)) {     
1970 >       >       rc = agg_extent_add(agg_entry, entry, &new_recx);               
1971 >       >       return rc;                                                      
1972 >       }                   

When there are just one ext [1 ~ 6M], the initial agg_entry->ae_cur_stripe.as_extent_cnt is zero right? so it will directly call agg_extent_add()

Reply:
hmm, you are right.